### ISSUE TITLE
Possible port to PC

### ISSUE THREAD

**Issue by AntonioND** [neo-engine/neo#23]
I was wondering if you have thought about splitting the engine part and the hardware part, and doing a port to PC using SDL2 or something similar. It would be useful for testing purposes, for example.

**Comment by PH111P** [comment]
That definitely sounds like a good idea! While, splitting off everything HW-related will take some time and lots of efforts, I think this is indeed feasible.

**Comment by AntonioND** [comment]
Of course, I was as vague as possible on purpose, this is no easy task. 😅

**Comment by Srchronotrigger** [comment]
I'm not demanding or asking anything, i'm not saying that is easy to do, but since you two are talking about ports it would be cool to think about the idea of porting to the 3DS natively, in a way it's the natural way, as it already runs on NDS.

Another interesting platform is the Playstation 2, it's a personal interest of mine, but I would find it very interesting to see some NDS pokemon maps running on the Playstation obviously excluding the possibility of the touchscreen. I like to see hardware challenges, to explore some console limits.

Finally, it's a suggestion not related to ports, but to the project itself, I could even create an FR for it, have you thought about creating a support for an unofficial "wonder trade"? it would be necessary to implement this function on the client-side and also create a server-side (like minecraft or other hostable games) to allow the trade between players, on the client-side the user could put the IP address of the desired host and then connect into that specific room and trading pokemons with those connected to the same host and this function could be cross-platform if the ports are done.

By the way congratulations for the project, it is very interesting and has a lot of potential.

**Comment by PH111P** [comment]
Hi @Srchronotrigger,

porting the game to different platforms is definitely something one should think about; that would require (perhaps a lot of) time though.

- For a 3DS port, one would have to adapt some graphics (in particular for the top screen) to the different screen ratios and one would lose access to the GBA slot; however, the generally better hardware of the 3DS might make this worthwhile (and this shouldn't take too much time in total).

- I think while a PS2 port does sound interesting, it would require a lot of work (and time) though.

- Regarding wonder trade/GTS functionality, this is indeed on my rather long list of things that I want to implement at some point. There are also other fan projects that have such a functionality, it might make sense to think about interoperability between these projects.

**Comment by Srchronotrigger** [comment]
I see, I would like to have the ability to contribute to some projects than just suggesting things, there are many interesting projects like yours that deserve a community effort, it's not fair to demand everything from a single programmer, and what you have done is a lot thing.

about the current version, I tried to play on 3DS via Twilight menu v25.7.0 but the game only renders some texts with a message (uistring failed), on melonDS it runs normally, I know the focus is not on the 3DS, but I'm reporting it anyway.

**Comment by PH111P** [comment]
I will try to look into the issue with Twilight (that is actually something that is supposed to work), could you please open a separate issue for that one?
(The "uistring failed" message typically means that some file system related things don't work properly; which build of the game were you trying to run?)

**Comment by Srchronotrigger** [comment]
Of course, i will open an issue for detail the problem, i have tested the 1.0.1-alpha, i will test other versions and take some pictures and look if the problem persist.
